torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -749,4 +749,78 @@ xla::XlaOp BuildMaskedScatter(xla::XlaOp input, xla::XlaOp mask,
       scatter_dnums);
 }
 
+std::vector<xla::XlaOp> BuildAmpForeachNonFiniteCheckAndUnscale(
+    const std::vector<xla::XlaOp>& inputs, const xla::XlaOp& found_inf_float,
+    const xla::XlaOp& inv_scale) {
+  const xla::PrimitiveType origin_type =
+      XlaHelpers::ShapeOfXlaOp(found_inf_float).element_type();
+  std::vector<xla::XlaOp> found_infs;
+  xla::XlaOp one = xla::One(inputs[0].builder(), xla::PrimitiveType::PRED);
+  for (size_t i = 0; i < inputs.size(); ++i) {
+    xla::XlaOp all_finite =
+        xla::ReduceAll(xla::IsFinite(inputs[i]), one,
```

Comment:
Regarding my previous comment about support of scalar one, I assumed you meant here? `AllReduce` is reduction across all cores where `ReduceAll` is reduction of a tensor within a core. I did not see the usage of `AllReduce` in this pr. Could you give me a more detail example, like if non-scalar one is given, where the execution error will happen and what would the error message looks like?

Reply:
Sorry for the typo in the previous rely. Yes, it is `ReduceAll`. And I checked the new implementation by using `found_inf` as a non-scalar tensor, it ran successfully. So please disregard my previous comment about the scalar or non-scalar usage.

Reply:
@yaochengji Does that mean https://github.com/pytorch/pytorch/pull/48570/files#diff-b17e8e6fc2ce9966e0a0b1a7d322a2a5938deb671b2e6f1c471ae1a5937a4dd4R144 can be removed? ;) Thanks!

Reply:
Yeah, I have reverted it back.